R/mash_wrapper.R

```diff
@@ -492,70 +521,61 @@ load_multitrait_R_sumstat <- function(
   out <- list(bhat = bhat, sbhat = sbhat)
-
+  
   # Check if variants are the same in both bhat and sbhat
   if (!identical(out$bhat$variants, out$sbhat$variants)) {
     stop("Error: Variants in bhat and sbhat are not the same.")
   }
-  var_idx <- 1:nrow(out$bhat)
+  var_idx = 1:nrow(out$bhat)
   if (!is.null(filter_file)) {
-    variants <- out$bhat$variants
-    var_idx <- split_variants_and_match(variants, filter_file, max_rows_selected)
+    variants = out$bhat$variants
+    var_idx = split_variants_and_match(variants, filter_file, max_rows_selected)
   }
-
+  
   if (top_loci) {
-    union_top_loci <- merge_susie_cs(susie_fit, coverage, complementary)
-    if (!is.null(union_top_loci)) {
-      strong_signal_df <- union_top_loci %>%
-        group_by(credible_set_names) %>%
-        filter(median_pip == max(median_pip)) %>%
-        slice(1) %>%
-        ungroup()
-      var_idx <- which(out$bhat$variants %in% strong_signal_df$variant_id)
+    union_top_loci <- merge_susie_cs(susie_fit,coverage)
+    if(!is.null(union_top_loci)){
+    strong_signal_df <- union_top_loci %>% 
+                 group_by(credible_set_names) %>%
+                 filter(median_pip == max(median_pip)) %>%
+                 slice(1) %>% ungroup()
+    var_idx <- which(out$bhat$variants %in% strong_signal_df$variant_id)
     } else {
       var_idx <- NULL
     }
   }
 
-
+  
   # Extract only subset of data
   variants <- out$bhat$variants[var_idx]
-  out$bhat <- out$bhat[var_idx, , drop = FALSE]
-  out$sbhat <- out$sbhat[var_idx, , drop = FALSE]
-
-  if (nan_remove) {
-    out <- handle_invalid_summary_stat(out, bhat = "bhat", sbhat = "sbhat", z = nan_remove)
-  }
-  rownames(out$bhat) <- rownames(out$sbhat) <- variants
-  colnames(out$bhat)[2:ncol(out$bhat)] <- colnames(out$sbhat)[2:ncol(out$bhat)] <- trait_names
-  out$bhat <- out$bhat[, -which(names(out$bhat) == "variants"), drop = FALSE]
-  out$sbhat <- out$sbhat[, -which(names(out$sbhat) == "variants"), drop = FALSE]
-  out$region <- names(susie_fit)
-
-  if (length(exclude_condition) > 0) {
-    if (all(exclude_condition %in% colnames(out$bhat))) {
-      out$bhat <- out$bhat[, -exclude_condition, drop = FALSE]
-      out$sbhat <- out$sbhat[, -exclude_condition, drop = FALSE]
-    } else {
-      # Handle the case where exclude_condition names do not exist in column
-      # names of dat This could be an error
+  out$bhat <- out$bhat[var_idx,,drop=FALSE]
+  out$sbhat <- out$sbhat[var_idx,,drop=FALSE]
+  
+   rownames(out$bhat) <- rownames(out$sbhat) <- variants
+   colnames(out$bhat)[2:ncol(out$bhat)] <- colnames(out$sbhat)[2:ncol(out$bhat)] <- trait_names
+   out$bhat <- out$bhat[,-which(names(out$bhat)=="variants"), drop = FALSE]
+   out$sbhat <- out$sbhat[,-which(names(out$sbhat)=="variants"), drop = FALSE]
+   out$region = names(susie_fit)
+                    
+   if (length(exclude_condition) > 0) {
```

Comment:
as I suggested earlier, you can make `exclude_condition = NULL` by default and here use `is.null` to check it.